src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -109,6 +112,9 @@ public class SelectStatement implements CQLStatement.SingleKeyspaceCqlStatement
     private static final NoSpamLogger noSpamLogger = NoSpamLogger.getLogger(SelectStatement.logger, 1, TimeUnit.MINUTES);
 
     public static final int DEFAULT_PAGE_SIZE = 10000;
+    public static final String TOPK_CONSISTENCY_LEVEL_ERROR = "Top-k queries can only be run with consistency level one. Consistency level %s was used.";
```

Comment:
I think we are missing the client warning saying that vector indexes are experimental. That warning message should probably be quite similar to`SASIIndex.USAGE_WARNING` and `View.USAGE_WARNING`. We could also include information about limitations. For example:
```
SAI ANN indexes on vector columns are experimental and are not recommended for production use. 
They cannot be queried with a read consistency level higher than ONE/LOCAL_ONE. 
They cannot use paging either, nor be used in aggregation queries. 
These limitations are planned to be removed in future versions.
```